core/i18n/src/ITwinLocalization.ts

```diff
@@ -51,25 +50,32 @@ export class ITwinLocalization implements Localization {
       caches: [],
       ...options?.detectorOptions,
     };
+
     this._initOptions = {
       interpolation: { escapeValue: true },
       fallbackLng: "en",
+      maxRetries: 1, // prevents wasted time and potential timeouts when requesting localization files throws an error
       backend: this._backendOptions,
       detection: this._detectionOptions,
       ...options?.initOptions,
     };
 
     this.i18next
       .use(options?.detectorPlugin ?? i18nextBrowserLanguageDetector)
-      .use(options?.backendPlugin ?? XHR)
+      .use(options?.backendPlugin ?? Backend)
       .use(TranslationLogger);
   }
 
   public async initialize(namespaces: string[]): Promise<void> {
+    // In case defaultNS is passed into the constructor, make sure it's in the namespaces list
+    if (this._initOptions.defaultNS && namespaces.indexOf(this._initOptions.defaultNS) === -1) {
+      namespaces.push(this._initOptions.defaultNS);
+    }
+
     const initOptions: InitOptions = {
-      ... this._initOptions,
       ns: namespaces,
       defaultNS: namespaces[0],
+      ...this._initOptions,
```

Comment:
is this right? spreading at the end? you will overrite ns & defaultNS if defined in _initOptions

Reply:
You can pass both a list of namespaces and a default namespace in the class constructor. I did this spreading at the end intentionally so that if a default namespace `defaultNS` is passed into the class constructor, it doesn't get overridden by `namespaces[0]`. Though, perhaps I _shouldn't_ allow `ns` (a list of namespaces) to be overridden this way. I could join the list passed in the constructor with the list passed into this function?

Reply:
I've changed this now so that the spreading doesn't override anything. Also, if there was one passed in, the defaultNS passed into the constructor if used. Also, I combine the list of namespaces optionally passed into the constructor and into the initialize() function into one list that gets used; before, any namespaces passed into the constructor would've been overridden.